### DISCUSSION TITLE
verify that element exist in database

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46572] [Q&A] [answered]
**Hello everyone please how can I verify that an elemnt exist in databse ?**
Thank you.

**Comment by DylanKas** [comment]
What do you mean ? Can't you check your database ?
If for some reason you cannot access it, maybe dumping some result from your repository to find what you want ?

**Reply by Elmatbaisaad** [reply]
i created this query 
![image](https://user-images.githubusercontent.com/101296742/172555753-34693d73-5868-484a-b452-e1e3e2ecefbd.png)
So i want to verify in my controller if query returns null (no element with this specification in database)
and after verifying if its null i can add an elemnt else i pass

**Reply by Asmitta-01** [reply]
By default if the query finds nothing it will returns `null`. it is different in your project ?

**Comment by lyrixx** [comment, marked as answer]
I usually does something like:

```php
public function isAlreadyExist(string $email): bool
{
    return (bool) $this
        ->createQueryBuilder('i')
        ->select('COUNT(1)')
        ->andWhere('i.email = :email')->setParameter('email', $email)
        ->andWhere('i.acceptedAt IS NULL')
        ->getQuery()
        ->getSingleScalarResult()
    ;
}
```
